core/scheduler/scheduler.go

```diff
@@ -495,30 +495,46 @@ func newSlotTicker(ctx context.Context, eth2Cl eth2wrap.Client, clock clockwork.
 		return nil, err
 	}
 
-	chainAge := clock.Since(genesis)
-	height := int64(chainAge / slotDuration)
-	startTime := genesis.Add(time.Duration(height) * slotDuration)
-
-	resp := make(chan core.Slot)
+	currentSlot := func() core.Slot {
+		chainAge := clock.Since(genesis)
+		slot := int64(chainAge / slotDuration)
+		startTime := genesis.Add(time.Duration(slot) * slotDuration)
+
+		return core.Slot{
+			Slot:          slot,
+			Time:          startTime,
+			SlotsPerEpoch: int64(slotsPerEpoch),
+			SlotDuration:  slotDuration,
+		}
+	}
 
+	var (
+		resp = make(chan core.Slot)
+		slot = currentSlot()
+	)
 	go func() {
 		for {
-			resp <- core.Slot{
-				Slot:          height,
-				Time:          startTime,
-				SlotsPerEpoch: int64(slotsPerEpoch),
-				SlotDuration:  slotDuration,
+			select {
+			case <-ctx.Done():
+				return
+			case <-clock.After(slot.Time.Sub(clock.Now())):
```

Comment:
it'd be good to add tests for `newSlotTicket`

Reply:
we have tests for scheduler